2016-10-30  Manas Kumar  <kumar@lnxsrv06.seas.ucla.edu>

        * timespec.h (timespec_cmp): Original code had problem where extreme
        values of a and b would cause overflow in return value. Replace code
        with that using only conditional statements.

--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ coreutils-with-bug1/lib/timespec.h	2016-10-30 12:37:08.053294000 -0700
@@ -45,8 +45,11 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_sec < b.tv_sec)
+    return -1;
+  else if (a.tv_sec > b.tv_sec)
+    return 1;
+  else return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
